documents/extension/funding/service.go

```diff
@@ -175,23 +175,45 @@ func (s service) DeriveFromPayload(ctx context.Context, req *clientfunpb.Funding
 	var fd Data
 	fd.initFundingFromData(req.Data)
 
-	model, err := s.GetCurrentVersion(ctx, identifier)
+	fd.BorrowerId = req.Data.BorrowerId
+	fd.FunderId = req.Data.FunderId
+
+	funderID, err := identity.NewDIDFromString(req.Data.FunderId)
 	if err != nil {
-		apiLog.Error(err)
-		return nil, documents.ErrDocumentNotFound
+		return nil, err
 	}
-
-	ca, err := documents.FromClientCollaboratorAccess(req.ReadAccess, req.WriteAccess)
+	borrowerID, err := identity.NewDIDFromString(req.Data.BorrowerId)
 	if err != nil {
 		return nil, err
 	}
-
+	model, err := s.GetCurrentVersion(ctx, identifier)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
 	attributes, err := createAttributesList(model, fd)
 	if err != nil {
 		return nil, err
 	}
 
-	err = model.AddAttributes(ca, true, attributes...)
+	// check which Id needs to be added as a new collaborator to the document
+	var c []identity.DID
+	for _, id := range []identity.DID{funderID, borrowerID} {
```

Comment:
dont need to do this check. Documents automatically checks for duplicates. Just creat a `CollaboratorAccess` with funder and borrower set to ReadAccess. And it should work as expected

Reply:
It is not about duplicates.  As far as I understood it, it is the case: if you add a `funderID` in the data which is not a collaborator, yet on the document.


Both funder and borrower need WriteAccess for adding the signatures

Reply:
The correct usage of the rest api would be to add the funder as a collaborator and again add his   DID in the `data.funderID` of the funding agreement